lib/blockchain/chain.js

```diff
@@ -855,6 +855,37 @@ class Chain extends AsyncEmitter {
 
     assert(fork, 'No free space or data corruption.');
 
+    // Check NLR (No Large Reorganization)
+    if  (tip.height - fork.height >= this.nlrLimit) {
+      if (this.nlrLimit !== 0) {
+        this.logger.warning(
+          'NLR Activated: current=%h(%d) fork=%h(%d) reorg_size=%d',
+          tip.hash,
+          tip.height,
+          competitor.hash,
+          competitor.height,
+          tip.height - fork.height
+        );
+
+        const entryWithMostWork = (tip.height > competitor.height) ? tip : competitor
```

Comment:
isn't this always going to be `competitor`?

Reply:
No, the work value of a block depends upon the difficulty not only the height

`const entryWithMostWork = tip.chainwork.gt(competitor.chainwork) ? tip : competitor`

Reply:
okay I ended up switching this to just using the `competitor` instead of `entryWithMostWork`. Assuming that the walk down always happens from the longer chain which the competitor should always be if a reorg was triggered.

Here too, if the walk down is done on `pindexMostWork` and pindexMostWork is the tip of the incoming fork being considered, `competitor` is the equivalent.

https://github.com/floblockchain/flo/commit/2fe045194caf8137878128ebf01461b9a99e7fb4#diff-24efdb00bfbe56b140fb006b562cc70bR2328